ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/SurrogateTemplateIdCache.scala

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http.dbbackend
+
+import com.codahale.metrics.MetricRegistry
+import com.daml.caching.SizedCache
+import com.daml.http.dbbackend.Queries.SurrogateTpId
+import com.daml.metrics.Metrics
+import org.slf4j.LoggerFactory
+
+object SurrogateTemplateIdCache {
+  private val logger = LoggerFactory.getLogger(classOf[SurrogateTemplateIdCache])
+  private val maxSize = 1000L
```

Comment:
open to suggestions on what maxSize should be in terms of how many templates we expect

Reply:
I think this is probably something you want to allow to configure.

Reply:
I can make it an env variable (although not sure what general opinion on having env variables sprinkled around in code is) but in either case it would still be good to have a sensible default

Reply:
Sure. Regarding the environment variable, at this level I would recommend you just accept a parameter for the class to be instantiated and thread this through the JSON API class that instantiates this. Right now we get all of our configuration parameters from the CLI and we should stick to that. Regarding the sensible default, what is the measurement unit? Bytes, entries, something else?

Reply:
the unit is `entries` , so eviction will kick in after 1000 entries currently which maybe a tad high.

Reply:
It depends a lot on the size of the entries and the ledger setup, we can have 1000 very small packages or 10 huge ones, or a mix of both. I would say that an acceptable cautious value to provide as a default could be 10. Do you think it would be feasible to replace the `SizedCache` with a `WeightedCache` ([code](https://github.com/digital-asset/daml/blob/648acf52f8fa7a82cbb1d4523ed4ca384f2e1e32/ledger/caching/src/main/scala/com/daml/caching/WeightedCache.scala)) and give the user control over the maximum size in bytes?

Reply:
ok just read this , I'd like to keep it as `SizedCache` for now I've lowered the num entries to `10` I mean worse case is we are dealing with more evictions and the value should now be configurable now. Happy to re-visit this when we have more information.